core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1227,12 +1220,19 @@ class DAGScheduler(
                   }
                 }
                 submitWaitingChildStages(shuffleStage)
+              } else {
+                // Some tasks had failed; let's resubmit this shuffleStage
+                // TODO: Lower-level scheduler should also deal with this
+                logInfo(s"Resubmitting $shuffleStage (${shuffleStage.name}) " +
+                  "because some of its tasks had failed: " +
+                  shuffleStage.findMissingPartitions().mkString(", "))
```

Comment:
@srowen That's what I was going to do, but didn't know how much would get approved :) Thanks for warm welcome.